src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -255,11 +255,9 @@ internal static ulong ParseUInt64(ReadOnlySpan<char> value, NumberStyles styles,
             return result;
         }
 
-        private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberStyles styles, ref NumberBuffer number, NumberFormatInfo info)
+        private static bool TryParseNumber(ref ReadOnlySpan<char> str, NumberStyles styles, ref NumberBuffer number, NumberFormatInfo info)
         {
             Debug.Assert(str != null);
```

Comment:
Do we still need this? ReadOnlySpan is a struct

Reply:
No, this will always be false.

Reply:
(nit)Remove the assert?